server/plugins/messageStorage/sqlite.ts

```diff
@@ -198,6 +251,116 @@ class SqliteMessageStorage implements SearchableMessageStorage {
 		});
 	}
 
+	async fetch_rollbacks(since_version: number) {
+		const res = await this.serialize_fetchall(
+			`select version, rollback_forbidden, statement
+			from rollback_steps
+			join migrations on migrations.id=rollback_steps.migration_id
+			where version > ?
+			order by version desc, step asc`,
+			since_version
+		);
+		const result: Rollback[] = [];
+
+		// convert to Rollback[]
+		// requires ordering in the sql statement
+		res.reduce((combined: Rollback[], cur) => {
```

Comment:
Is this reduction just being used as a for loop? This is pretty hard for me to follow.

Reply:
Yes.
There I was, being pleased with myself that I'm a real JS programmer that uses all the functional thingies....

okayokay, I'm gonna rewrite it with a plain boring loop then :wink:

Reply:
done